test/test_torch.py

```diff
@@ -16140,17 +16140,26 @@ def test_addmm_sizes(self, device, dtype):
         for m in [0, 1, 25]:
             for n in [0, 1, 10]:
                 for k in [0, 1, 8]:
+                    if False and dtype.is_complex:  # bug to be fixed in another PR
+                        alpha = 1.2 + 0.3j
+                        beta = 0.5 + 1.6j
+                    else:
+                        alpha = 1.2
+                        beta = 0.8
                     M = torch.randn(n, m, device=device, dtype=dtype)
                     m1 = torch.randn(n, k, device=device, dtype=dtype)
                     m2 = torch.randn(k, m, device=device, dtype=dtype)
-                    res1 = torch.addmm(M, m1, m2)
-                    res2 = torch.zeros(n, m, device=device, dtype=dtype)
-                    res2 += M
+                    res1 = torch.addmm(M, m1, m2, alpha=alpha, beta=beta)
+                    res2 = torch.full_like(res1, math.nan)
+                    torch.addmm(M, m1, m2, alpha=alpha, beta=beta, out=res2)
+                    res3 = (beta * M).cpu().tolist()
                     for i in range(n):
```

Comment:
rather than doing this in a loop, maybe better compare with numpy matmul?

Reply:
If we compare with numpy, this means we will be unable to run this test without numpy

Reply:
That's OK. We will likely soon require NumPy to run the tests, anyway.